### DISCUSSION TITLE
Change from flannel to cilium in existing cluster

### DISCUSSION THREAD

**Discussion by primeXchange** [kube-hetzner/terraform-hcloud-kube-hetzner#802] [Q&A]
Is it possible to change the CNI from flannel to cilium in an existing cluster? I want to make use of the Egress feature.

Is there anything I'd need to do besides setting the `cni_plugin = "cilium"` variable?

**Comment by mysticaltech** [comment]
@primeXchange I would imagine it's delicate to do that. But you could try!

Either way, you can use Velero to backup and restore your whole cluster (while changing the underlying CNI and upgrading to the latest and greatest version of the kube-hetzner module if you want).
